### DISCUSSION TITLE
How/Can Serverpod fit into a microservices architecture?

### DISCUSSION THREAD

**Discussion by israelws** [serverpod/serverpod#645] [General]
1. Greate work !!!
2. Can we build a microservice architecture using Serverpod? do you have an example? 

thanks,

**Comment by ajay-k07** [comment]
Yes I believe You can build microservice architecture using serverpod. And you can also communicate with different services using traditional http or you can use messaging queue like pub/sub or you can use serverpod built-in streams and messaging option. but few things you have to take care like naming server, service discovery ,load-balancing. but you can use pre-defined solutions for this

**Comment by vlidholt** [comment]
We are building better support for this right now! There may be certain features (future calls, keeping a state on the server) that you cannot use if you run Serverpod on e.g. Google Cloud Run, but it can still be a great option.

**Reply by binemmanuel** [reply]
I'm considering building an App with Serverpod and I had this same question that led me here. What's the update on this?

**Comment by dufguix** [comment]
Why do you need microservices ?

What I see is :
- Serverpod is stateless. so horizontal scaling is possible
- Serverpod has modules. so you can "cluster" your code
